internal/stack/serverless.go

```diff
@@ -273,10 +281,24 @@ func (sp *serverlessProvider) BootUp(options Options) error {
 		printUserConfig(options.Printer, config)
 	}
 
-	logger.Infof("Starting local agent")
-	err = sp.startLocalAgent(options, config)
+	logger.Infof("Starting local services")
+	err = sp.startLocalServices(options, config)
 	if err != nil {
-		return fmt.Errorf("failed to start local agent: %w", err)
+		return fmt.Errorf("failed to start local services: %w", err)
+	}
+
+	if settings.LogstashEnabled {
+		//Re-create clients if running on an existing project
+		if sp.kibanaClient == nil {
+			if err = sp.createClients(project); err != nil {
+				return err
+			}
+		}
+		// Updating the output with ssl certificates created in startLocalServices
+		err = project.UpdateLogstashFleetOutput(sp.profile, sp.kibanaClient)
+		if err != nil {
+			return err
+		}
 	}
```

Comment:
What happens if the `elastic-package stack up -d --provider serverless` runs more than once ? 

This project already exists and it will have the Logstash output updated.

Would it be better to add a new variable to manage the scenario where the project has being created ? 

```go
	var project *serverless.Project
	projectCreated := false
	project, err = sp.currentProject(config)
	switch err {
	default:
		return err
	case serverless.ErrProjectNotExist:
		logger.Infof("Creating %s project: %q", settings.Type, settings.Name)
		config, err = sp.createProject(settings, options, config)
		if err != nil {
			return fmt.Errorf("failed to create deployment: %w", err)
		}
		project, err = sp.currentProjectWithClientsAndFleetEndpoint(config)
		if err != nil {
			return fmt.Errorf("failed to retrieve latest project created: %w", err)
		}

		logger.Infof("Creating agent policy")
		err = project.CreateAgentPolicy(options.StackVersion, sp.kibanaClient, settings.LogstashEnabled)

		if err != nil {
			return fmt.Errorf("failed to create agent policy: %w", err)
		}
		projectCreated = true
		// TODO: Ensuring a specific GeoIP database would make tests reproducible
		// Currently geo ip files would be ignored when running pipeline tests
	case nil:
		logger.Debugf("%s project existed: %s", project.Type, project.Name)
		printUserConfig(options.Printer, config)
	}

	logger.Infof("Starting local services")
	err = sp.startLocalServices(options, config)
	if err != nil {
		return fmt.Errorf("failed to start local services: %w", err)
	}

	if projectCreated && settings.LogstashEnabled {
		//Re-create clients if running on an existing project
		if sp.kibanaClient == nil {
			if err = sp.createClients(project); err != nil {
				return err
			}
		}
		// Updating the output with ssl certificates created in startLocalServices
		err = project.UpdateLogstashFleetOutput(sp.profile, sp.kibanaClient)
		if err != nil {
			return err
		}
	}
```

Reply:
@mrodm Sounds good. Just one case that I am wondering.

Is there a possibility that the tls certificates are updated when we run `elastic-package stack up -d --provider serverless` ?

Reply:
Tested this locally and the client certificate is updated every time `elastic-package stack up -d --provider serverless` is run and hence the `elastic-agent` certs need to be updated in fleet.

Reply:
Yes, it could be that the certs are updated when starting local services (docker-compose), since that function executes `applyServerlessResources`.

If certs are updated, then this should be executed always. Good catch!

Reply:
> Tested this locally and the client certificate is updated every time `elastic-package stack up -d --provider serverless` is run and hence the `elastic-agent` certs need to be updated in fleet.

Btw, this should not happen, certificates should be regenerated only when they are not valid. I guess that `checkExpectedCertUsage` is failing for these now?